### ISSUE TITLE
Next_Observation and Reward clamping in MOPO

### ISSUE THREAD

**Issue by ssimonc** [polixir/NeoRL#5]
Dear Authors,
In your MOPO implementation, when generating the transitions from the ensemble, you take under consideration the min/max of the training batch, as follows:
```python
obs_max = torch.as_tensor(train_buffer['obs'].max(axis=0)).to(self.device)
obs_min = torch.as_tensor(train_buffer['obs'].min(axis=0)).to(self.device)
rew_max = train_buffer['rew'].max()
rew_min = train_buffer['rew'].min()
```
So that, prior to the computation of the penalised reward and the addition of the experience tuple in the batch, you can clamp the observation and the reward between the min and the max recovered above:
```python
next_obs = torch.max(torch.min(next_obs, obs_max), obs_min)
reward = torch.clamp(reward, rew_min, rew_max)
```
Is there a particular reason behind this choice? I could not find a correspondence in the original MOPO implementation/publication, or is it simply due to other re-implementation needs, considering the different framework used?

Kind regards

**Comment by IcarusWizard** [comment]
Hi, @ssimonc. Thanks for your question. During the re-implementation of MOPO, we find sometimes the ensemble will generate unrealistic samples that are very far away from the data support which leads to unstable training behaviors. We introduce the clip trick to stabilize the algorithms for the benchmark. Although it is not presented in the original MOPO implementation, we do find a similar trick in MoReL, which is another popular Model-based offline RL algorithm.

**Comment by ssimonc** [comment]
Many thanks for the immediate reply. 
I can see how it can help stabilise the training process, luckily my concerns for other reasons were unfounded.
Thank you also for the MoReL suggestion, I'll for sure dive deeper into its publication.
